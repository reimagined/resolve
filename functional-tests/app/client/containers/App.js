import React from 'react'
import { Helmet } from 'react-helmet'

const App = ({ staticPath }) => {
  const stylesheetLink = {
    rel: 'stylesheet',
    type: 'text/css',
    href: `${staticPath}/bootstrap.min.css`,
  }
  const faviconLink = {
    rel: 'icon',
    type: 'image/png',
    href: `${staticPath}/favicon.ico`,
  }
  const links = [stylesheetLink, faviconLink]
  const meta = {
    name: 'viewport',
    content: 'width=device-width, initial-scale=1',
  }

  return (
    <div>
      <Helmet title="Functional Tests App" link={links} meta={[meta]} />
      <h2 align="center">Basic tests</h2>
      <h2 align="center">View model tests</h2>
    </div>
  )
}

export default App
