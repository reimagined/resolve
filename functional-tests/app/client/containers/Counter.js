import React, { useState, useEffect } from 'react'
import { useViewModel, useCommand } from 'resolve-react-hooks'

const aggregateId = 'counter-aggregate-id'

const Counter = () => {
  const [counter, setCounter] = useState(0)

  const { connect, dispose } = useViewModel(
    'counter',
    [aggregateId],
    setCounter
  )

  const increaseCounter = useCommand({
    type: 'increase',
    aggregateId,
    aggregateName: 'Counter',
    payload: counter,
  })

  const decreaseCounter = useCommand({
    type: 'decrease',
    aggregateId,
    aggregateName: 'Counter',
    payload: counter,
  })

  useEffect(() => {
    connect()
    return () => {
      dispose()
    }
  }, [])

  return (
    <div>
      <button onClick={increaseCounter}>+</button>
      <div id="counter">{counter}</div>
      <button onClick={decreaseCounter}>-</button>
    </div>
  )
}

export default Counter
