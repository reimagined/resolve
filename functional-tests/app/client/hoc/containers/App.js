import React from 'react'
import { Helmet } from 'react-helmet'

const App = ({ staticPath }) => {
  const stylesheetLink = {
    rel: 'stylesheet',
    type: 'text/css',
    href: `${staticPath}/bootstrap.min.css`,
  }
  const faviconLink = {
    rel: 'icon',
    type: 'image/png',
    href: `${staticPath}/favicon.ico`,
  }
  const links = [stylesheetLink, faviconLink]
  const meta = {
    name: 'viewport',
    content: 'width=device-width, initial-scale=1',
  }

  return (
    <div>
      <Helmet title="HOC client" link={links} meta={[meta]} />
      <h2 align="center">HOC client</h2>
    </div>
  )
}

export default App
