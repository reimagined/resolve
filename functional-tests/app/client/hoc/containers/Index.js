import React from 'react'

class Index extends React.PureComponent {
  render() {
    return <div className="example-wrapper">HOC components tests</div>
  }
}

export default Index
