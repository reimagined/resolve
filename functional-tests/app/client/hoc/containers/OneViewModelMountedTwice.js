import React from 'react'
import { connect } from 'react-redux'
import { connectViewModel, sendAggregateAction } from 'resolve-redux'
import { Button } from 'react-bootstrap'

class OneViewModelMountedTwice extends React.PureComponent {
  render() {
    const { isLoading, data, salt } = this.props

    if (isLoading !== false) {
      return null
    }

    return <h3>{`likes${salt}:${data.likes}`}</h3>
  }
}

const mapStateToOptions = (state, ownProps) => {
  const aggregateId = ownProps.userId

  return {
    viewModelName: 'cumulative-likes',
    aggregateIds: [aggregateId],
    aggregateArgs: {},
  }
}

const mapStateToProps = () => ({})
const mapDispatchToProps = () => ({})

const ConnectedLikesA = connectViewModel(mapStateToOptions)(
  connect(mapStateToProps, mapDispatchToProps)(OneViewModelMountedTwice)
)
const ConnectedLikesB = connectViewModel(mapStateToOptions)(
  connect(mapStateToProps, mapDispatchToProps)(OneViewModelMountedTwice)
)

class Wrapper extends React.PureComponent {
  like = () => this.props.likeUser(this.props.match.params.id)

  render() {
    const userId = this.props.match.params.id

    return (
      <div className="example-wrapper">
        <Button onClick={this.like}>like</Button>
        <ConnectedLikesA userId={userId} salt="#1" />
        <ConnectedLikesB userId={userId} salt="#2" />
      </div>
    )
  }
}

const mapWrapperDispatchToProps = (dispatch) => ({
  likeUser: (userId) =>
    dispatch(sendAggregateAction('user', 'like', userId, {})),
})

export default connect(() => ({}), mapWrapperDispatchToProps)(Wrapper)
