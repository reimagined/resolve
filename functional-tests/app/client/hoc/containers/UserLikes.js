import React from 'react'
import { connect } from 'react-redux'
import { connectViewModel } from 'resolve-redux'
import { ListGroup, ListGroupItem } from 'react-bootstrap'

class UserLikes extends React.PureComponent {
  render() {
    const { isLoading, data } = this.props

    if (isLoading !== false) {
      return null
    }

    return (
      <div className="example-wrapper">
        <ListGroup className="example-list">
          {Object.keys(data).map((userId) => (
            <ListGroupItem key={userId}>
              {`${userId}:${data[userId]}`}
            </ListGroupItem>
          ))}
        </ListGroup>
      </div>
    )
  }
}

const mapStateToOptions = (state, ownProps) => {
  const aggregateId = ownProps.userId

  return {
    viewModelName: 'cumulative-likes',
    aggregateIds: [aggregateId],
    aggregateArgs: {},
  }
}

const mapStateToProps = () => ({})
const mapDispatchToProps = () => ({})

const ConnectedLikesA = connectViewModel(mapStateToOptions)(
  connect(mapStateToProps, mapDispatchToProps)(UserLikes)
)
const ConnectedLikesB = connectViewModel(mapStateToOptions)(
  connect(mapStateToProps, mapDispatchToProps)(UserLikes)
)

class Wrapper extends React.PureComponent {
  render() {
    const userId = this.props.match.params.id

    return (
      <div className="example-wrapper">
        <ConnectedLikesA userId={userId} />
        <ConnectedLikesB userId={userId} />
      </div>
    )
  }
}

export default Wrapper
