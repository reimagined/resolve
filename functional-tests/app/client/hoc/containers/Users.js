import React from 'react'
import { connectReadModel } from 'resolve-redux'
import { connect } from 'react-redux'
import { Button } from 'react-bootstrap'

class Users extends React.PureComponent {
  render() {
    const { users } = this.props

    if (!users) {
      return null
    }

    return (
      <div className="example-wrapper">
        {users.map(({ id }) => (
          <div key={id}>
            <Button
              href={`users-likes/${id}`}
              key={id}
            >{`likes of ${id}`}</Button>
            <br />
          </div>
        ))}
      </div>
    )
  }
}

export const mapStateToOptions = () => ({
  readModelName: 'users',
  resolverName: 'all',
  resolverArgs: {},
})

export const mapStateToProps = (state, { data, isLoading }) => {
  return {
    isLoading,
    users: data,
  }
}

export const mapDispatchToProps = () => ({})

export default connectReadModel(mapStateToOptions)(
  connect(mapStateToProps, mapDispatchToProps)(Users)
)
