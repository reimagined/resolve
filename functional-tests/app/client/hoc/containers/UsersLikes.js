import React from 'react'

class UsersLikes extends React.PureComponent {
  render() {
    return <div className="example-wrapper">Users likes</div>
  }
}

export default UsersLikes
