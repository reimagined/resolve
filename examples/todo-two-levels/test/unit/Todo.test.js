import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router';

import { Todo } from '../../client/containers/Todo';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <StaticRouter location="/id1" context={{}}>
        <Todo
          todos={{
            id1: {
              text: 'text1',
              checked: true
            },
            id2: {
              text: 'text2',
              checked: false
            }
          }}
          createItem={() => {}}
          toggleItem={() => {}}
          removeItem={() => {}}
          aggregateId={() => {}}
        />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
