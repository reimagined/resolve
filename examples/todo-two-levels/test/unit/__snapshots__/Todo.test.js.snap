// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-default"
    >
      <div
        className="container"
      >
        <p
          className="navbar-text"
        >
          <img
            className=""
            src="/resolve-logo.png"
          />
           Todo Two Levels Example
        </p>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
        >
          <p
            className="navbar-text navbar-right"
          >
            <a
              className="navbar-link"
              href="https://facebook.com/resolvejs/"
            >
              <img
                className=""
                src="/fb-logo.png"
              />
            </a>
          </p>
          <p
            className="navbar-text navbar-right"
          >
            <a
              className="navbar-link"
              href="https://twitter.com/resolvejs"
            >
              <img
                className=""
                src="/twitter-logo.png"
              />
            </a>
          </p>
          <p
            className="navbar-text navbar-right"
          >
            <a
              className="navbar-link"
              href="https://github.com/reimagined/resolve"
            >
              <img
                className=""
                src="/github-logo.png"
              />
            </a>
          </p>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="example-wrapper"
  >
    <form
      className="form-inline"
    >
      <a
        aria-current="true"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          className="example-arrow-button"
          src="/left-arrow-button.png"
        />
        <span
          className="example-back-label"
        >
          Back
        </span>
      </a>
      <div
        className="example-task-name"
      >
        Task's List
      </div>
    </form>
    <ul
      className="example-list list-group"
    >
      <li
        className="list-group-item"
      >
        <label
          className="checkbox-inline"
          title=""
        >
          <input
            checked={true}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          text1
        </label>
        <img
          className="example-close-button"
          onClick={[Function]}
          src="/close-button.png"
        />
      </li>
      <li
        className="list-group-item"
      >
        <label
          className="checkbox-inline"
          title=""
        >
          <input
            checked={false}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          text2
        </label>
        <img
          className="example-close-button"
          onClick={[Function]}
          src="/close-button.png"
        />
      </li>
    </ul>
    <form
      className="example-form form-inline"
    >
      <input
        className="example-form-control form-control"
        onKeyPress={[Function]}
        placeholder="New Task"
        type="text"
      />
      <button
        className="example-button btn btn-success"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Add Task
      </button>
    </form>
  </div>
</div>
`;
